cli_tools/common/image/importer/request.go

```diff
@@ -103,6 +104,8 @@ type ImageImportRequest struct {
 	Timeout               time.Duration   `name:"timeout" validate:"required"`
 	UefiCompatible        bool
 	Zone                  string `name:"zone" validate:"required"`
+	DiskName              string
```

Comment:
This is creating a leaky abstraction, where the caller needs to understand when to provide `DiskName` and when to provide `ImageName`. My intuition is the safer alternative is to not provide this field, and instead determine the PD's URI by looking at the return value when you call `Inflate`. What do you think?

Reply:
I thought about it, but I just wanted to save the URIs of the created disks in the same order like the filesURIs and as requests, like what is done in the multi_image_importer before.
I can use the same way of naming disks like inflater (exectionID + idx) [here](https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/cli_tools/common/image/importer/inflater.go#L309) to get the disk name but I feel like it's not good idea because it will fail if the inflater changed it's default way of naming disks. what do u think?

Reply:
To maintain order, can't you do something like this?

```
	for i, request := range requests {
		req := request
		disk, err := disk.NewDisk(req.Project, req.Zone, req.DiskName)
		if err != nil {
			return nil, err
		}

		disks = append(disks, disk)
		logPrefix := fmt.Sprintf("[import-%s]", req.DaisyLogLinePrefix)

		group.Go(func() error {
                        // Use result of r.singleDiskImporter.Import to update disks[i]
			return r.singleDiskImporter.Import(ctx, req, r.logger.NewLogger(logPrefix))
		})
	}
```

Reply:
Done.